### ISSUE TITLE
Server error when Azure Dev Deploy finished

### ISSUE THREAD

**Issue by v-jiaodi** [Azure/azure-dev#865]
**Describe the issue:**
Server error when Azure Dev Deploy finished.

**Repro Steps:**
Run command: ` bash ./test-templates.sh -c false -t <template_name> -b pr/551 `.

**Error Message:**

![image](https://user-images.githubusercontent.com/80496810/195049039-181a6ecd-cc56-44ae-9782-0ef39b8d4707.png)

Through investigation, we found that it was caused by API service exception

![image](https://user-images.githubusercontent.com/80496810/195053031-032cd936-8b41-4218-a15b-d62e4fa9b5e9.png)

![image](https://user-images.githubusercontent.com/80496810/195050764-55310d70-6921-4df4-a946-4b5bd2f79c21.png)

**Environment:**
OS: Windows desktop, Linux Desktop, WSL ,Mac ,Codesapce and DevContainer.
Templates:
https://github.com/Azure-Samples/todo-nodejs-mongo-swa-func
https://github.com/Azure-Samples/todo-python-mongo-swa-func
Branch: pr/551

**Expected behavior:**
The api server is normal, and e2e test is pass.

@jongio for notification.

**Comment by rajeshkamal5050** [comment]
@jongio @madebygps did we make any changes to other templates as part of #551? any changes in your PR could cause this?

**Comment by jongio** [comment]
Yes, I will investigate.

**Comment by v-xuto** [comment]
@jongio We do the manual test about template (`todo-python-mongo-swa-func` and `todo-python-mongo`), also encountered the same error when Azure Dev Deploy finished.

**Error:**
![image](https://user-images.githubusercontent.com/57166602/196145401-e17801a6-8255-4627-a2c0-18a4940b4f21.png)

**Environment:**
OS: Windows desktop
Templates:
https://github.com/Azure-Samples/todo-python-mongo-swa-func
https://github.com/Azure-Samples/todo-python-mongo
Branch: [pr/892](https://github.com/Azure/azure-dev/pull/892)
Azd version: `azd version 0.3.0-beta.3-daily.1921004 (commit e52ea013fede3560b90f4cf9d0a8e6d6ceac376a)`

**Comment by v-xuto** [comment]
@jongio We do the manual test about template (`todo-python-mongo`), also encountered the same error when Azure Dev Deploy finished.

**Error:**
![image](https://user-images.githubusercontent.com/57166602/196145401-e17801a6-8255-4627-a2c0-18a4940b4f21.png)

**Environment:**
OS: Windows desktop
Templates:
https://github.com/Azure-Samples/todo-python-mongo
Branch: [pr/892](https://github.com/Azure/azure-dev/pull/892)
Azd version: `azd version 0.3.0-beta.3-daily.1921004 (commit e52ea013fede3560b90f4cf9d0a8e6d6ceac376a)`

**Comment by jongio** [comment]
Please try again with main.  If it fails, please leave it in failed state and provide me and @weikanglim with a link to the env.

**Comment by zedy-wj** [comment]
@jongio - We have tried again with main, it is still unstable. When we execute the command `bash ./test-templates.sh -c false -t todo-python-mongo`, the playwright test failed. At this point, when we click the [api site](https://app-api-4lfkbyaf7ycse.azurewebsites.net/), it reported `Application Error`. After a period of time, the api service can be deployed successfully, and the error will not occur again when `playwright test` is executed again.

In summary, the error was caused by the `api service` is not deployed successfully yet, the playwright test is run first.

**Comment by jongio** [comment]
Can you try with a different region and see if you get the same error?

**Comment by zedy-wj** [comment]
@jongio - We have tried with region `East US`, and get the same error. This time it was due to a failed deployment, you can reach from this [link](https://app-api-335oni2tqecwu.azurewebsites.net/).
Template: https://github.com/Azure-Samples/todo-python-mongo
Branch: main
azd version: 0.3.0-beta.3-daily.1937328 (commit d18003a6799b545e4050544f9fa05f1eaedb0948)

**Comment by weikanglim** [comment]
The detector was extremely helpful. I found the following errors related to the docker container crashing:

```
File "<frozen importlib._bootstrap>", line 973, in _find_and_load_unlocked
2022-10-24T03:24:54.032196245Z ModuleNotFoundError:
No module named 'uvicorn'2022-10-24T03:24:54.032199945Z ]
```

**Comment by jongio** [comment]
What do you mean by the detector?

**Comment by jongio** [comment]
Could be related: https://github.com/Azure/azure-dev/issues/932

**Comment by rajeshkamal5050** [comment]
Not seeing these template failures in the pipelines. Might be a temporary region issue. Reopen if you see again.

**Comment by zedy-wj** [comment]
@rajeshkamal5050, @jongio  - In today's test, the issue persists and occurs at a high rate.

OS: Windows desktop.
azd version: `azd version 0.3.0-beta.6-pr.1947898 (commit a7b9ee6038c4ecbe8ddda4c0ba1acbd2ec208fc8)`
branch: [pr/996](https://github.com/Azure/azure-dev/pull/996)
templates:
[https://github.com/azure-samples/todo-python-mongo](https://github.com/azure-samples/todo-pytho-n-mongo)
https://github.com/azure-samples/todo-csharp-cosmos-sql 
https://github.com/Azure-Samples/todo-csharp-sql 
https://github.com/Azure-Samples/todo-java-mongo 
https://github.com/azure-samples/todo-nodejs-mongo

**Comment by jongio** [comment]
Fixed with #996
